src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -210,31 +210,59 @@ private static CastResult TryGet(nuint source, nuint target)
             if (obj != null)
             {
                 MethodTable* mt = RuntimeHelpers.GetMethodTable(obj);
-                nuint interfaceCount = mt->InterfaceCount;
+                nint interfaceCount = mt->InterfaceCount;
                 if (interfaceCount != 0)
                 {
                     MethodTable** interfaceMap = mt->InterfaceMap;
-                    for (nuint i = 0; ; i += 4)
+                    if (interfaceCount >= 4)
```

Comment:
```suggestion
                    if (interfaceCount < 4)
```
?

Reply:
Ah ha! This may explain the strange benchmark results 😅